src/components/Graph/index.tsx

```diff
@@ -75,10 +85,21 @@ const MemoizedGraph = React.memo(function Layout({
 
   const handleNodeClick = React.useCallback(
     (e: React.MouseEvent<SVGElement>, props: NodeProps) => {
-      setSelectedNode(props.properties.text);
-      openModal();
+      if (navigationMode) {
+        const subTree = searchSubTree(mainTree, props.id);
+
+        if (subTree.length) {
+          const flatTree = flattenTree(subTree);
```

Comment:
flattenTree from `json-editor-parser.ts` is applied to the sub tree whose node is clicked by the user. Here you apply the same function applied on line 64  of this file into a subtree. FRACTAL.